actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -422,17 +424,21 @@ def encode_params(params)
             @param_encoder.call(params)
           end
 
+          def parse_body(body)
+            @response_parser.call(body)
+          end
```

Comment:
Would be nice if this was memoized. Frequent to have a bunch of checks against response.parsed_body['key1'], response.parsed_body['key2'].